Holesky rescue

nflaig opened the pull request:
See https://github.com/ChainSafe/lodestar/issues/7503

nflaig committed:
Bump default memory limit to 32768

nflaig committed:
Add more holesky bootnodes

nflaig committed:
Enable connectToDiscv5Bootnodes by default

nflaig committed:
Increase default slotImportTolerance to 3200 slots

matthewkeil committed:
feat: move blacklisting code onto single commit

nflaig committed:
Enable forceCheckpointSync by default

nflaig committed:
Remove default bootnodes

twoeths committed:
fix: model AttesterSlashing without bigint (#7502)

**Motivation**

- @nflaig 's node was down due to OOM, could be because of processing a
bunch of AttesterSlashing
- in the past we used to have memory issue with it, see
https://github.com/ChainSafe/lodestar/issues/6112

**Description**

-  model AttesterSlashing without BigInt

Co-authored-by: Tuyen Nguyen <twoeths@users.noreply.github.com>

twoeths committed:
fix: prune checkpoint states in processState()

twoeths commented:
@nflaig @matthewkeil just merged https://github.com/ChainSafe/lodestar/pull/7505

nflaig committed:
Reduce slot import tolerance

nflaig committed:
Do not force checkpoint sync by default

wemeetagain committed:
feat: handle chain stall in peer manager

twoeths committed:
feat: prune persisted cp states (#7510)

nflaig committed:
Revert "fix: model AttesterSlashing without bigint (#7502)"

This reverts commit e19edb22dde0247f288de2f56827f8e541fc0648.

nflaig committed:
Merge remote-tracking branch 'origin/cayman/starved-churn' into holesky-rescue

nflaig committed:
Increase default max cp states on disk to 100 before pruning

nflaig committed:
Revert slot import tolerance to sane value

nflaig committed:
chore: use native compute proposer/sync committee

From https://github.com/ChainSafe/lodestar/pull/7499

(squash merged)

twoeths committed:
fix: findSeedStateToReload PR#7516

twoeths committed:
fix: correct log

wemeetagain committed:
feat: improve dumpPeer(s) APIs

twoeths committed:
fix: checkpoint file datastore on docker (#7525)

nflaig committed:
Increase default targetPeers to 200

twoeths committed:
feat: use local checkpoint state for holesky-rescue (#7509)

**Motivation**

- we already have local checkpoint states in either db or file, this PR
use one of them to save syncing time the next time the node is restarted
- see
https://github.com/ChainSafe/lodestar/issues/7504#issuecomment-2689547378

**Description**

2 new init unfinalized state options:

- new `--lastPersistedCheckpointState`, true by default to load from the
last safe persisted checkpoint state. And the last safe persisted
checkpoint state is considered unfinalized
- use `--checkpointState` option: this is old option and support
finalized state only, this PR supports booting from unfinalized state
similar to `--lastPersistedCheckpointState` ==> to quickly sync a new
node, we can use this option. Then remove it and the next time node will
use `--lastPersistedCheckpointState` option by default
- we can configure one of our nodes with
`chain.nHistoricalStatesFileDataStore = true`
- then feed other nodes with a persisted "safe checkpoint state" from
there in `~/checkpoint_states` folder

- a persisted checkpoint state is consider to be safe to boot if
  - it should be the checkpoint state that's unique in its epoch
- its last processed block slot should be at epoch boundary or last slot
of previous epoch
   - state slot should be at epoch boundary
   - state slot should be equal to epoch * SLOTS_PER_EPOCH

other options to boot from `stateArchived` or `checkpointSyncUrl` are
considered finalized states including `wssCheckpoint`. It's not possible
to use `wssCheckpoint` option with unfinalized state for now.

**TODO**
- this is for `holesky-rescue`, consider supporting this for `unstable`
branch too
- update document in that case

---------

Co-authored-by: Tuyen Nguyen <twoeths@users.noreply.github.com>

twoeths committed:
feat: holesky-resuce - get persisted checkpoint state api (#7541)

**Motivation**

- implement an api to get a node synced asap

**Description**

- new api: `eth/v1/lodestar/persisted_checkpoint_state` to return a
state based on an optional `rootHex:epoch` param
- if not specified, return the latest safe checkpoint state
- a node need to specify `--checkpointState` from the previous PR #7509

**Test**
- [x] `curl -H "Accept: application/octet-stream"
http://localhost:9596/eth/v1/lodestar/persisted_checkpoint_state -o
latest_checkpoint_state.ssz`
- [x] `curl -H "Accept: application/octet-stream"
http://localhost:9596/eth/v1/lodestar/persisted_checkpoint_state?checkpoint_id=0x4f4d4c1b81141fe77a4a1c6a376dbe64ed9baa8f123664195e4f710c9fc4238d:118936
-o state_epoch_118936.ssz`

---------

Co-authored-by: Tuyen Nguyen <twoeths@users.noreply.github.com>
Co-authored-by: Nico Flaig <nflaig@protonmail.com>

nflaig committed:
fix: ensure we only receive SSZ checkpoint states if REST API is queried (#7542)

We have added a new api in
https://github.com/ChainSafe/lodestar/pull/7541 to serve latest
checkpoint states. However the problem is that `--checkpointState` right
now assumes that we receive a ssz-serialized state bytes, so we also
need to ensure that the REST API only returns SSZ responses by applying
the correct Accept header.


This PR includes the following changes
- Ensure we only receive SSZ responses if REST API is queried
- Consider `--forceCheckpointSync` flag and if set always prioritze
remote checkpoint from `--checkpointState` over latest safe checkpoint
in db
- Minor tweaks to logs to make it easier to understand what the node is
doing
- Misc formatting changes

nflaig committed:
fix: use correct subnet when validating gossip attestations (#7543)

Fix a bug in gossip validation of electra attestations

twoeths committed:
feat: track oppool insert outcome for api attestation separately

nflaig commented:
Holesky has been rescued
